### ISSUE TITLE
Narrator: Entries in Workspace window are read as class names

### ISSUE THREAD

**Issue by MikhailArkhipov** [Microsoft/RTVS#3171]
Narrator reads `Microsoft.R.Component.ConnectionManager.Implementation.ViewModel...`

should be reading 'Microsoft R Client (3.3.2.0)` (connection name)

**Comment by MikhailArkhipov** [comment]
If you override ToString() on the model, narrator will read it

**Comment by huguesv** [comment]
Or this link (which I've posted on teams), which shows how to do it in XAML
https://blogs.msdn.microsoft.com/gautamg/2010/03/10/how-to-get-automation-working-properly-on-data-bound-wpf-list-or-combo-box/

**Comment by AlexanderSher** [comment]
This bug in it's original form isn't valid anymore, cause with fixed tab navigation you can't select the item entry - only it's buttons (cause item itself is no different from connect button). Though for them there is a bunch of changes required:
- Narrator should read "Local" and "Remote" toggles correctly
- Buttons should contain the name of connection and action and shouldn't read the tooltip
- Pressing "Enter" shouldn't start the connection until "Connect" button is selected

**Comment by MikhailArkhipov** [comment]
Actually tooltip usually a good information. This is how VS buttons are read. 'Enter' can start connection, why? Default action can be anything. You can make 'Space' to open editing, then tab can move around eit fields. 'Enter' then closes and connects, Escape cancels. I believe you can check other trees for the typical patterns - Team Explorer or Solution Explore.

**Comment by MikhailArkhipov** [comment]
https://github.com/Microsoft/RTVS/pull/3207

**Comment by AlexanderSher** [comment]
> 'Enter' can start connection, why? Default action can be anything. 

Because we already have a "Connect" button that is part of the entry and is selected as a first element inside the entry. There are 2 options: 
- allow "Connect" button to be focused and then we don't assign any action on the entry itself, cause it duplicates the button
- allow entry to be focused or accessed by TAB, assign default action to it and then skip "Connect" button

**Comment by AlexanderSher** [comment]
> Actually tooltip usually a good information. This is how VS buttons are read.

Yes, but in case of "Edit" and "Delete" buttons the same string is used for tooltip and `AutomationProperties.Name`. I can replace that with URL and command line arguments (similar to "Connect" button)

**Comment by MikhailArkhipov** [comment]
Enter is for keyboard access, button is for the mouse. I think it is too much to require people to focus on connect before hitting Enter

**Comment by AlexanderSher** [comment]
> I think it is too much to require people to focus on connect before hitting Enter

Exactly. That's why either "Connect" button or entry itself should be focusable, but not both.
